src/Raven.Server/Documents/Queries/QueryMetadata.cs

```diff
@@ -1516,6 +1517,9 @@ private static QueryFieldName ExtractFieldNameFromArgument(QueryExpression argum
 
             if (argument is ValueExpression value) // escaped string might go there
                 return new QueryFieldName(value.Token, value.Value == ValueTokenType.String);
+            
+            if (argument is MethodExpression method && method.ToString() == Constants.Documents.Indexing.Fields.DocumentIdFieldName) //id property might be written as id()
```

Comment:
Let's avoid the allocation here, we can do this without calling `ToString` by comparing directly.